third_party/amd/lib/TritonAMDGPUToLLVM/SPMDOpToLLVM.cpp

```diff
@@ -25,10 +27,50 @@ struct GetNumProgramsOpConversion
   }
 };
 
+struct CondBarrierOpConversion
+    : public ConvertOpToLLVMPattern<triton::amdgpu::condBarrierOp> {
+  using ConvertOpToLLVMPattern<
+      triton::amdgpu::condBarrierOp>::ConvertOpToLLVMPattern;
+
+  LogicalResult
+  matchAndRewrite(triton::amdgpu::condBarrierOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Location loc = op->getLoc();
+    auto i32ty = rewriter.getIntegerType(32);
+    auto workIDX = rewriter.create<ROCDL::ThreadIdXOp>(loc, i32ty);
+    auto constZero = rewriter.create<LLVM::ConstantOp>(
+        loc, i32ty, IntegerAttr::get(i32ty, 0));
+    auto constWarpSize = rewriter.create<LLVM::ConstantOp>(
+        loc, i32ty, IntegerAttr::get(i32ty, 256));
```

Comment:
This is not warp size per se? It's more a "batch size" or something that all threads (that can belong to multiple warps) are barriered together? Also we don't want to just hardcode 256 here. What about making it an attribute to the op?

Reply:
That's absolutely right. Original design is to have a boolean input arg but the code stays hacky because of some issues I had before. I'll fix it.